atc/api/accessor/claims_cacher.go

```diff
@@ -49,13 +49,10 @@ func (c *claimsCacher) GetAccessToken(rawToken string) (db.AccessToken, bool, er
 		return db.AccessToken{Token: rawToken, Claims: entry.claims}, true, nil
 	}
 
-	token, found, err := c.accessTokenFetcher.GetAccessToken(rawToken)
+	token, _, err := c.accessTokenFetcher.GetAccessToken(rawToken)
```

Comment:
Checking `found` broke caching which is why this commit ignores it. I might dig into this again with a clearer head. Maybe the test is wrong?

Reply:
Yeah, looks like I was a bit lazy when I wrote this test, and a bit negligent when I wrote the code. This diff fixes the test and adds a missing test for the `!found` case

```diff
diff --git a/atc/api/accessor/claims_cacher.go b/atc/api/accessor/claims_cacher.go
index ffb5d6916..aa0daff41 100644
--- a/atc/api/accessor/claims_cacher.go
+++ b/atc/api/accessor/claims_cacher.go
@@ -49,10 +49,13 @@ func (c *claimsCacher) GetAccessToken(rawToken string) (db.AccessToken, bool, er
 		return db.AccessToken{Token: rawToken, Claims: entry.claims}, true, nil
 	}
 
-	token, _, err := c.accessTokenFetcher.GetAccessToken(rawToken)
+	token, found, err := c.accessTokenFetcher.GetAccessToken(rawToken)
 	if err != nil {
 		return db.AccessToken{}, false, err
 	}
+	if !found {
+		return db.AccessToken{}, false, nil
+	}
 	payload, err := json.Marshal(token.Claims)
 	if err != nil {
 		return db.AccessToken{}, false, err
diff --git a/atc/api/accessor/claims_cacher_test.go b/atc/api/accessor/claims_cacher_test.go
index 90f959667..206ee0a7b 100644
--- a/atc/api/accessor/claims_cacher_test.go
+++ b/atc/api/accessor/claims_cacher_test.go
@@ -29,11 +29,26 @@ var _ = Describe("ClaimsCacher", func() {
 	})
 
 	It("fetches claims from the DB", func() {
-		claimsCacher.GetAccessToken("token")
+		expectedToken := db.AccessToken{Claims: db.Claims{Username: "foo"}}
+		fakeAccessTokenFetcher.GetAccessTokenReturns(expectedToken, true, nil)
+
+		token, found, err := claimsCacher.GetAccessToken("token")
+		Expect(err).ToNot(HaveOccurred())
+		Expect(found).To(BeTrue())
+		Expect(token).To(Equal(expectedToken))
+
 		Expect(fakeAccessTokenFetcher.GetAccessTokenCallCount()).To(Equal(1), "did not fetch from DB")
 	})
 
+	It("returns not found if the token isn't found", func() {
+		fakeAccessTokenFetcher.GetAccessTokenReturns(db.AccessToken{}, false, nil)
+		_, found, err := claimsCacher.GetAccessToken("token")
+		Expect(err).ToNot(HaveOccurred())
+		Expect(found).To(BeFalse())
+	})
+
 	It("doesn't fetch from the DB when the result is cached", func() {
+		fakeAccessTokenFetcher.GetAccessTokenReturns(db.AccessToken{}, true, nil)
 		claimsCacher.GetAccessToken("token")
 		claimsCacher.GetAccessToken("token")
 		Expect(fakeAccessTokenFetcher.GetAccessTokenCallCount()).To(Equal(1), "did not cache claims")
```